src/user.fz

```diff
@@ -198,11 +198,26 @@ module user (module base_dir String) is
       full0 := "{first} {last}".trim
       full0.is_empty ? email : full0
 
+    validate_mail(str String) =>
+      s := str.split "@" .as_array
+      if s.count != 2
+        logger.log "illegal mail address: {str}"
+        false
+      # NYI: UNDER DEVELOPMENT: allow non ASCII mail addresses?
+      else if !(str.utf8 ∀ b->
+          # ASCII but no space and no control character
+          0x21 <= b <= 0x7E)
+        logger.log "illegal mail address: >{str}<"
+        false
+      else
+        true
```

Comment:
email validation is technically more complicated than this, see https://e-mail.wtf/

it's probably fine for now